redhat-appstudio/remote-secret
# fix: Secret data cycles when 2 RemoteSecrets have the same target

### What does this PR do?
Bugfix of $TITLE

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-674

### How to test this PR?

1. create 2 RemoteSecrets
```shell
kubectl apply -f - <<EOF
apiVersion: appstudio.redhat.com/v1beta1
kind: RemoteSecret
metadata:
  name: test-rs
spec:
  targets:
  - namespace: default
    secret:
      name: rs-with-overrides-1
stringData:
  username: blabla
---
apiVersion: appstudio.redhat.com/v1beta1
kind: RemoteSecret
metadata:
  name: test-rs-2
spec:
  targets:
  - namespace: default
    secret:
      name: rs-with-overrides-1
stringData:
  username: abcd
EOF
```
2. ensure `test-rs-2` has error in target:
![image](https://github.com/redhat-appstudio/remote-secret/assets/73115616/ca5e4f73-cd60-43b5-bd17-8137c4d18ed7)

3. look at `default/rs-with-overrides-1` secret and ensure the labels and annotations/data do not cycle.
